airflow/www/views.py

```diff
@@ -4512,17 +4507,14 @@ def process_form(self, form, is_created):
                 del form.extra
         del extra_json
 
-        for key in self.extra_fields:
+        for key, field_name in form.iter_extra_field_names():
```

Comment:
`extra_fields` is removed since it is simply `extra_field_name_mapping.keys()`, but since Python 3.7 the dict maintains ordering on its own the extra list is not needed anymore.

`extra_field_name_mapping` is delibrately moved to ConnectionForm since Flask-Appbuilder has some annoying introspecting mechanism that tries to resolve values on the view class too eagerly. Moving the property to the form class ensures it is only calculated when needed.

Note that the form does not even cache the value; the providers manager already does the caching, and an additional cache layer providers minimal benefit.